### PR TITLE
MRG: Add talairach.xfm and MRI scaling to coreg

### PR DESCRIPTION
**Pull request by larsoner**
~~This should work once https://github.com/nipy/nibabel/issues/619 is sorted out.~~

@christianbrodbeck does this look good to you assuming I can work out the `nibabel` bug and thus get tests to pass?

### CODE REVIEW

**Comment by larsoner** [comment]
Okay transform issue fixed. Ready for review/merge from my end.

**Comment by larsoner** [comment]
... and I did a tiny bit of plotting in `freeview` and the scaled MRIs seemed to plot properly. It would be good if someone could cross-check to make sure.

**Comment by larsoner** [comment]
@christianbrodbeck WDYT?

**Comment by christianbrodbeck** [comment]
Conceptually it makes sense, for reviewing it will take me a little longer to get to it (sorry)...

**Comment by larsoner** [comment]
@cjayb any experience working with MRIs? If so do you want to take a look?

@christianbrodbeck any time to look this week? We're trying to push out a release and it would be good to have this if possible.

File: mne/tests/test_coreg.py [hunk 1]
```diff
@@ -149,7 +223,7 @@ def test_fit_matched_points():
 
     # test exceeding tolerance
     tgt_pts[0, :] += 20
-    assert_raises(RuntimeError, fit_matched_points, tgt_pts, src_pts, tol=10)
+    pytest.raises(RuntimeError, fit_matched_points, tgt_pts, src_pts, tol=10)
```
Comment (massich) [inline mne/tests/test_coreg.py]: I would use the new syntactic
```py

**Comment by massich** [comment]
LGTM

**Comment by christianbrodbeck** [comment]
Looking into it now.

File: mne/coreg.py [hunk 2]
```diff
@@ -561,6 +568,14 @@ def _find_mri_paths(subject, skip_fiducials, subjects_dir):
                 dup.append(pformat(surf_fname, name=name))
     del surf_dup_name, name, path, dup, hemi
 
+    # transform files (talairach)
+    paths['transforms'] = []
+    transform_fname = os.path.join(mri_transforms_dirname, 'talairach.xfm')
+    path = transform_fname.format(subjects_dir=subjects_dir, subject=subject)
+    if os.path.exists(path):
+        paths['transforms'].append(transform_fname)
+    del transform_fname, path
```
Comment (christianbrodbeck) [inline mne/coreg.py]: Is this `del` statement left over from debugging?

File: mne/coreg.py [hunk 3]
```diff
@@ -1091,3 +1120,124 @@ def scale_source_space(subject_to, src_name, subject_from=None, scale=None,
...
+        combine_transforms(
+            combine_transforms(
+                combine_transforms(
+                    T_ras_mri, invert_transform(S), 'ras', 'mri'),
+                F_mri_ras, 'ras', 'ras'),
+            F_ras_mni, 'ras', 'mni_tal')
+    _write_fs_xfm(fname_to, T_ras_mni['trans'], kind)
+
+
+def _get_sf(subject_to, subject_from, scale, subjects_dir):
```
Comment (christianbrodbeck) [inline mne/coreg.py]: This kind of duplicates the `_scale_params` function further up, maybe the two could be unified?

File: mne/coreg.py (hunk 2 above)
Comment (larsoner) [reply mne/coreg.py]: No, it's more for code safety to ensure if `path` is used later it's using the correct one

**Comment by larsoner** [comment]
That plot looks incorrect. Can you see if `freeview` shows it correctly? It would tell us if there is a bug in `plot_bem` or in this code

**Comment by christianbrodbeck** [comment]
I am not familiar with freeview, how would you plot it?

**Comment by christianbrodbeck** [comment]
Okay yeah, adding the T1 under Volumes and smoothwm under Surfaces looks like it has the same mismatch.

**Comment by larsoner** [comment]
Here is an example for `sample`:
```
freeview -v mri/T1.mgz -f bem/inner_skull.surf:edgecolor=red -f bem/outer_skull.surf:edgecolor=orange -f bem/outer_skin.surf:edgecolor=yellow --layout 4
```
screenshot from 2018-04-26 12-09-24

You'll need to make sure the `.surf` files are in your subject before morphing, though. Otherwise there will be extra work involved to extract them from the scaled `-bem.fif` output to `.surf` format (in `mm` IIRC).

**Comment by larsoner** [comment]
Okay there was an ordering problem that should now be fixed. This is what I see using 90/110/120 as my `sample` scale factors:
```
$ freeview -v mri/T1.mgz -f bem/inner_skull.surf:edgecolor=red -f bem/outer_skull.surf:edgecolor=orange -f bem/our_skin.surf:edgecolor=yellow --layout 4
```
screenshot from 2018-04-26 13-42-22

And then deforming `fsaverage` with the same scale factors and doing:
```Python
import mne
surf = mne.read_bem_surfaces(subjects_dir + '/sample_deformed/bem/sample_deformed-inner_skull-bem.fif')[0]
mne.write_surface(subjects_dir + '/sample_deformed/bem/inner_skull.surf',
                  surf['rr'] * 1000, surf['tris'])
mne.viz.plot_bem('sample_deformed', brain_surfaces='pial',
                 subjects_dir=subjects_dir)
```
I get:

figure_1


@christianbrodbeck can you see if things are fixed now for you, too?

**Comment by christianbrodbeck** [comment]
Yes it looks good now. Not sure how to do an equivalent test for the XFM file. Also, it looks like `viz.plot_bem` distorts image proportions?

screen shot 2018-04-26 at 9 26 21 pm
figure_1

**Comment by larsoner** [comment]
> Also, it looks like viz.plot_bem distorts image proportions?

It probably assumes that the `zooms` for each dimension are identical. This is a bug we should probably fix separately.

**Comment by agramfort** [comment]
Thx !

**Pull request merged**